src/main/java/at/favre/lib/crypto/HKDF.java

```diff
@@ -161,7 +191,22 @@ public static HKDF from(HkdfMacFactory macFactory) {
      * @return new byte array of output keying material (OKM)
      */
     public byte[] extractAndExpand(byte[] saltExtract, byte[] inputKeyingMaterial, byte[] infoExpand, int outLengthByte) {
-        return new Expander(macFactory).execute(new Extractor(macFactory).execute(saltExtract, inputKeyingMaterial), infoExpand, outLengthByte);
+        return extractAndExpand(macFactory.createSecretKey(saltExtract), inputKeyingMaterial, infoExpand, outLengthByte);
+    }
+
+    /**
+     * Convenience method for extract &amp; expand in a single method
+     *
+     * @param saltExtract         optional salt value (a non-secret random value);
+     * @param inputKeyingMaterial data to be extracted (IKM)
+     * @param infoExpand          optional context and application specific information; may be null
+     * @param outLengthByte       length of output keying material in bytes
+     * @return new byte array of output keying material (OKM)
+     */
+    public byte[] extractAndExpand(SecretKey saltExtract, byte[] inputKeyingMaterial, byte[] infoExpand, int outLengthByte) {
+        return new Expander(macFactory).execute(macFactory.createSecretKey(
```

Comment:
This answers my above question: yes, you use `macFactory` to wrap the `byte[]` into `SecretKey`

Reply:
Yes, but you can still use the old `byte[]` API